controllers/release/release_adapter_test.go

```diff
@@ -0,0 +1,414 @@
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appstudioshared "github.com/redhat-appstudio/managed-gitops/appstudio-shared/apis/appstudio.redhat.com/v1alpha1"
+	appstudiov1alpha1 "github.com/redhat-appstudio/release-service/api/v1alpha1"
+	"github.com/redhat-appstudio/release-service/tekton"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clientsetscheme "k8s.io/client-go/kubernetes/scheme"
+)
+
+var _ = Describe("Release Adapter", func() {
+	var (
+		adapter      *Adapter
+		cacheAdapter *Adapter
+
+		release             *appstudiov1alpha1.Release
+		releaseStrategy     *appstudiov1alpha1.ReleaseStrategy
+		releaseLink         *appstudiov1alpha1.ReleaseLink
+		applicationSnapshot *appstudioshared.ApplicationSnapshot
+		noCacheClient       client.Client
+
+		//kclient             client.Client
+	)
+
+	BeforeEach(func() {
+		ctx := context.Background()
+		noCacheClient, _ = client.New(cfg, client.Options{Scheme: clientsetscheme.Scheme})
+
+		releaseStrategy = &appstudiov1alpha1.ReleaseStrategy{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-releasestrategy-",
+				Namespace:    testNamespace,
+			},
+			Spec: appstudiov1alpha1.ReleaseStrategySpec{
+				Pipeline:              "release-pipeline",
+				Bundle:                "test-bundle",
+				Policy:                "test-policy",
+				PersistentVolumeClaim: "test-pvc",
+				ServiceAccount:        "test-account",
+			},
+		}
+		Expect(noCacheClient.Create(ctx, releaseStrategy)).Should(Succeed())
+
+		releaseLink = &appstudiov1alpha1.ReleaseLink{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "appstudio.redhat.com/v1alpha1",
+				Kind:       "ReleaseLink",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-releaselink-",
+				Namespace:    testNamespace,
+				Labels: map[string]string{
+					"release.appstudio.openshift.io/auto-release": "false",
+				},
+			},
+			Spec: appstudiov1alpha1.ReleaseLinkSpec{
+				DisplayName:     "Test ReleaseLink",
+				Application:     "test-app",
+				Target:          "default",
+				ReleaseStrategy: releaseStrategy.GetName(),
+			},
+		}
+		Expect(noCacheClient.Create(ctx, releaseLink)).Should(Succeed())
+
+		applicationSnapshot = &appstudioshared.ApplicationSnapshot{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-snapshot-",
+				Namespace:    "default",
+			},
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: testApiVersion,
+				Kind:       "ApplicationSnapshot",
+			},
+			Spec: appstudioshared.ApplicationSnapshotSpec{
+				Application: "testapplication",
+				DisplayName: "Test application",
+				Components:  []appstudioshared.ApplicationSnapshotComponent{},
+			},
+		}
+		Expect(noCacheClient.Create(ctx, applicationSnapshot)).Should(Succeed())
+
+		// adding the Type Metas as it loses it after creation
```

Comment:
Yeah, I think it would be nice to do it like you did here 
https://github.com/redhat-appstudio/release-service/blob/main/tekton/pipeline_run_test.go#L121